server/src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -67,16 +67,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   /** Stores a new {@link Message}. */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
     UserService userService = UserServiceFactory.getUserService();
     if (!userService.isUserLoggedIn()) {
       response.sendRedirect("/");
       return;
     }
 
     String user = userService.getCurrentUser().getEmail();
-    String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+    // response.setHeader("Refresh", "0; URL=http://localhost3000/userpage?user=" + user);
+
+    String userText = Jsoup.clean(request.getParameter("text"), Whitelist.basicWithImages());
+
+    String parsedText =
+        userText
+            .replace("[b]", "<strong>")
+            .replace("[/b]", "</strong>")
+            .replace("[i]", "<i>")
+            .replace("[/i]", "</i>");
+
+    String cleanedContent =
+        Jsoup.clean(parsedText, Whitelist.basicWithImages().addTags("strong").addTags("i"));
 
-    String userText = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+    userText = cleanedContent;
```

Comment:
You should just be able to write:

`userText = Jsoup.clean(parsedText, Whitelist.basicWithImages().addTags("strong").addTags("i"));